### PR TITLE
Add docs for working on lsif-typescript.

### PR DESCRIPTION
**Pull request by varungandhi-src**
### Test plan

n/a, only adding docs

### CODE REVIEW

File: Development.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Developing lsif-typescript
+
+## References
+
+- VS Code is a good reference point for the "correct" behavior
+  for code navigation functionality.
+- Keep a local checkout of Microsoft/TypeScript for checking out code examples.
```
Comment (Strum355) [inline Development.md:7]: Not recommending they use Sourcegraph to browse it :smh:

Comment (varungandhi-src) [reply Development.md:7]: Sourcegraph isn't as fast as ripgrep running on an SSD (few things are). 👁️

File: Development.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  so reading the code is the best way to understand how things work.
+
+## Running tests
+
+```sh
+npm run test
+```
+
+```
+npm run update-snapshots
```
Comment (Strum355) [inline Development.md:18]: Does this require having lsif-java installed?

Comment (varungandhi-src) [reply Development.md:18]: No, we have this logic in the test code. https://sourcegraph.com/github.com/sourcegraph/lsif-typescript/-/blob/src/main.test.ts?L101:10

Comment (Strum355) [reply Development.md:18]: Are there any other environment requirements we should document? Specific node version etc etc

**Review by Strum355 — APPROVED** [review]

File: Development.md (hunk 2 above)
Comment (varungandhi-src) [reply Development.md:18]: I don't think we need a specific node version, but we can add that later as a config file if needed (vs just having it in the docs).

**Pull request merged**